packages/components/src/timeline/index.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Card from '../card';
+import TimelineGroup from './timeline-group';
+
+export const groupByOptions = {
+	DAY: 'day',
+	WEEK: 'week',
+	MONTH: 'month',
+};
+
+export const orderByOptions = {
+	ASC: 'asc',
+	DESC: 'desc',
+};
+
+const Timeline = ( props ) => {
+	const { className, items, groupBy, orderBy } = props;
+
+	const timelineClassName = classnames( 'woocommerce-timeline', className );
+
+	// Early return in case no data was passed to the component.
+	if ( ! items || items.length === 0 ) {
+		return (
+			<Card title={ 'Timeline' } className={ timelineClassName }>
+				<p className="timeline_no_events">
+					{ __( 'No data to display', 'woocommerce-admin' ) }
+				</p>
+			</Card>
+		);
+	}
+
+	// Reducer used for grouping items.
+	const itemsToGroups = ( groups, item ) => {
+		const timeToMoment = ( ts ) => moment.unix( ts );
+		const haveSameMoment = ( g, i ) =>
+			timeToMoment( g.id ).isSame( timeToMoment( i.datetime ), groupBy );
+
+		const groupIndex = groups.findIndex( ( g ) =>
+			haveSameMoment( g, item )
+		);
+		if ( groupIndex < 0 ) {
+			return [
+				...groups,
+				{
+					id: item.datetime,
```

Comment:
Why are we switching from calling the data `datetime` to `id`? When reading the sorting function below, seeing `.id` being used confused me a little bit, but then I noted that ID is actually `datetime `.

Reply:
Good catch, I can see how that is confusing.

The reason I went with ID is that the datetime is used to identify which timeline group this is and where it fits in the overall timeline, and it didn't feel semantically correct to call it `datetime`. Your point is valid though; this is definitely confusing.

Do you have any suggestions here? I'd rather not call this `datetime` because this isn't really about the date; it's a way to identify this group and sort the list of groups.
Maybe `key` would be better?

Reply:
The only confusing part here is that it is used for sorting as well. I think in this case it would maybe be good to just add comment around the sorting by ID to identify the type of data?  Besides that, I understand your reasoning.

Reply:
I've refactored the code quite significantly, which includes just calling this `datetime` and then renaming the sorting function to `sortByDate`.

If you don't mind @dwainm could you take another look and let me know whether I just threw the readability out the window or if the code has been improved? 🙂

Reply:
Resolving this as per https://github.com/woocommerce/woocommerce-admin/pull/3614#discussion_r413855447